types/block_test.go

```diff
@@ -113,6 +113,44 @@ func TestBlockID(t *testing.T) {
 	}
 }
 
+// TestHeaderID probes the ID function of the BlockHeader type.
+func TestHeaderID(t *testing.T) {
```

Comment:
I'd probably throw the sanity check in validateHeaderAndBlock

all the test blocks is one thing, but all of the blocks in the whole consensus set is another level of assurance, especially because it's a sanity check that'll be run for every block that gets mined in all the integration tests as well